src/core/public/chrome/global_search/global_search_service.ts

```diff
@@ -46,16 +47,127 @@ export interface GlobalSearchCommand {
    * @param value search query
    * @param callback callback function when search is done
    */
-  run(value: string, callback?: () => void): Promise<ReactNode[]>;
+  run(value: string, callback?: () => void, abortSignal?: AbortSignal): Promise<ReactNode[]>;
```

Comment:
Can we update the function interface for better extensibility? in case there will be more context we will need to pass in the future
```suggestion
  run(value: string, options: {callback?: () => void, abortSignal?: AbortSignal}): Promise<ReactNode[]>;
```